app/services/from_address_creation.rb

```diff
@@ -78,10 +79,15 @@ def email_identity
 
   private
 
-  def saved_from_address
+  def saved_email_address
     @saved_from_address ||= FromAddress.find_by(
       service_id: from_address.service_id
-    ).email_address
+    )
+    if @saved_from_address.nil?
+      return nil
+    end
+
+    @saved_from_address.email_address
```

Comment:
```suggestion
    )&.email_address
```

I think the safe navigation operator should work here.

Reply:
Or we could pull out the db call into it's own method:

```
def saved_email_address
  return nil if saved_from_address.nil?
  
  saved_from_address.email_address
end

def saved_from_address
  @saved_from_address ||= FromAddress.find_by(
    service_id: from_address.service_id
  )
end
```

Reply:
Going to merge as is- I like the null coalesce but feels more elegant/less readable to me, I find the method readable and want to only return the email from the helper method rather than splitting it into a helper for the saved object then inspecting it or having two helpers, I just put a nil check in because it seems a remotely possible edge case if the form is old

TY!